docs/overview/elastic_request.md

```diff
@@ -0,0 +1,43 @@
+---
+id: overview_elastic_request
+title: "Elastic Request"
+---
+
+We can represent an Elasticsearch request as a generic data type that returns a value of type `A`. 
```

Comment:
Does not reflect the latest API in which the desired result type is provided upon retrieving the results of an executed request.